rcl_logging_interface/include/rcl_logging_interface/rcl_logging_interface.h

```diff
@@ -91,6 +95,59 @@ RCL_LOGGING_INTERFACE_PUBLIC
 RCUTILS_WARN_UNUSED
 rcl_logging_ret_t rcl_logging_external_set_logger_level(const char * name, int level);
 
+/// Get the logging directory.
+/**
+ * Uses various environment variables to construct a logging directory path.
+ *
+ * Use $ROS2_LOG_DIR if ROS2_LOG_DIR is set and not empty.
+ * Otherwise, use $ROS2_HOME/log, using ~/.ros for ROS2_HOME if not set or if empty.
+ *
+ * It also expands '~' to the current user's home directory.
+ *
+ * \param[inout] directory The C string pointer at which to write the directory path.
+ *   Only meaningful if the call is successful.
+ * \param[in] allocator The allocator to use for memory allocation.
+ * \return RCL_LOGGING_RET_OK if successful, or
+ * \return RCL_LOGGING_RET_ERROR if an unspecified error occurs.
+ */
+RCL_LOGGING_INTERFACE_PUBLIC
+inline
+rcl_logging_ret_t
+rcl_logging_get_logging_directory(const char ** directory, rcutils_allocator_t allocator)
+{
+  const char * log_dir_env;
+  const char * err = rcutils_get_env("ROS2_LOG_DIR", &log_dir_env);
+  if (!err && *log_dir_env != '\0') {
+    *directory = log_dir_env;
+  } else {
+    const char * ros2_home_dir_env;
+    err = rcutils_get_env("ROS2_HOME", &ros2_home_dir_env);
+    if (err || *ros2_home_dir_env == '\0') {
```

Comment:
If the environment variable was not set, `rcutils_get_env` still returns `err = NULL` and `ros2_home_dir_env=""`.
i.e.: if `err` is not `NULL` (error reported), the function should return an error and not try getting a different environment variable.

Reply:
looking at the implementation of `rcutils_get_env`, I understand. I applied the changes to both calls to `rcutils_get_env`: 4114f12a62239519559981dafec0b6b5b7113482